File: user_app/lib/features/models/address/address_payload.dart
import 'package:json_annotation/json_annotation.dart';

part 'address_payload.g.dart';

@JsonSerializable()
class CreateAddressPayload {
  final String label;
  final String type;
  final String? customLabel;
  final String fullName;
  final String phone;
  final String? alternatePhone;
  final String addressLine1;
  final String? addressLine2;
  final String? landmark;
  final String city;
  final String state;
  final String pincode;
  final String? country;
  final bool? isDefault;

  CreateAddressPayload({
    required this.label,
    required this.type,
    this.customLabel,
    required this.fullName,
    required this.phone,
    this.alternatePhone,
    required this.addressLine1,
    this.addressLine2,
    this.landmark,
    required this.city,
    required this.state,
    required this.pincode,
    this.country,
    this.isDefault,
  });

  factory CreateAddressPayload.fromJson(Map<String, dynamic> json) =>
      _$CreateAddressPayloadFromJson(json);
  Map<String, dynamic> toJson() => _$CreateAddressPayloadToJson(this);
}

@JsonSerializable()
class UpdateAddressPayload {
  final String? label;
  final String? type;
  final String? customLabel;
  final String? fullName;
  final String? phone;
  final String? alternatePhone;
  final String? addressLine1;
  final String? addressLine2;
  final String? landmark;
  final String? city;
  final String? state;
  final String? pincode;
  final String? country;
  final bool? isDefault;

  UpdateAddressPayload({
    this.label,
    this.type,
    this.customLabel,
    this.fullName,
    this.phone,
    this.alternatePhone,
    this.addressLine1,
    this.addressLine2,
    this.landmark,
    this.city,
    this.state,
    this.pincode,
    this.country,
    this.isDefault,
  });

  factory UpdateAddressPayload.fromJson(Map<String, dynamic> json) =>
      _$UpdateAddressPayloadFromJson(json);
  Map<String, dynamic> toJson() => _$UpdateAddressPayloadToJson(this);
}

@JsonSerializable()
class SetDefaultAddressPayload {
  final String addressId;

  SetDefaultAddressPayload({required this.addressId});

  factory SetDefaultAddressPayload.fromJson(Map<String, dynamic> json) =>
      _$SetDefaultAddressPayloadFromJson(json);
  Map<String, dynamic> toJson() => _$SetDefaultAddressPayloadToJson(this);
}

@JsonSerializable()
class DeleteAddressPayload {
  final String addressId;
  final bool permanent;

  DeleteAddressPayload({required this.addressId, this.permanent = false});

  factory DeleteAddressPayload.fromJson(Map<String, dynamic> json) =>
      _$DeleteAddressPayloadFromJson(json);
  Map<String, dynamic> toJson() => _$DeleteAddressPayloadToJson(this);
}

@JsonSerializable()
class RestoreAddressPayload {
  final String addressId;

  RestoreAddressPayload({required this.addressId});

  factory RestoreAddressPayload.fromJson(Map<String, dynamic> json) =>
      _$RestoreAddressPayloadFromJson(json);
  Map<String, dynamic> toJson() => _$RestoreAddressPayloadToJson(this);
}

File: user-web/src/lib/types/address/address.payload.ts
// @/lib/types/address/address.payload.ts

import { AddressLabelType } from "./address.enums";

export interface CreateAddressPayload {
  label: string;
  type: AddressLabelType;
  customLabel?: string;
  fullName: string;
  phone: string;
  alternatePhone?: string;
  addressLine1: string;
  addressLine2?: string;
  googleLocation?: string;
  landmark?: string;
  city: string;
  state: string;
  pincode: string;
  country?: string;
  formattedAddress?: string;
  location: {
    type: "Point";
    coordinates: [number, number]; // [longitude, latitude]
  };
  placeId?: number;
  osmType?: string;
  addressType?: string;
  isDefault?: boolean;
}

export interface UpdateAddressPayload {
  label?: string;
  type?: AddressLabelType;
  customLabel?: string;
  fullName?: string;
  phone?: string;
  alternatePhone?: string;
  addressLine1?: string;
  addressLine2?: string;
  landmark?: string;
  city?: string;
  state?: string;
  pincode?: string;
  country?: string;
  formattedAddress?: string;
  location?: {
    type?: "Point";
    coordinates?: [number, number];
  };
  placeId?: number;
  osmType?: string;
  addressType?: string;
  isDefault?: boolean;
}
